tools/prow-config-updater/github_commenter.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"knative.dev/pkg/test/ghutil"
+	"knative.dev/pkg/test/helpers"
+
+	"knative.dev/test-infra/tools/prow-config-updater/config"
+)
+
+const (
```

Comment:
I can see a huge unit test file just for this file

Reply:
I added unit test for `fileListCommentString`. But I don't see much value for adding unit tests for other functions..

Reply:
At least we should test calling a comment function and it returns value as expected, i.e. calling success function should not return failed comment

Reply:
I can change the function signature to also return the comment string, and check if the comment string is correct. Is it what you suggested? Just want to double check before I make the changes.

Reply:
I think there is a fake of ghutil, which you can use for checking the output. To do that I think you also need to make the interface in ghutil contain the new functions you added, let's add a TODO in this PR to add them later

Reply:
Added the TODO comment.